### PR TITLE
add prep-git.mk for pulling source from git repositories.

### PR DESCRIPTION
**Pull request by chincheta0815**
added prep-git.mk from github.com/oracle/solaris-userland.git and adapted prep.mk as done in solaris-userland.

### CODE REVIEW

**Comment by chincheta0815** [comment]
@pyhalov and @Mno-hime Would you be so kind and review the lines? I fell over some lines in prep.mk , e.g. $(eval $(call eval-git-rules)) that needed to be added to make the prep-git.mk work.
It calls a "define eval-git-rules" in the prep-git.mk.

I also added the ones defined in the prep-*.mk files that were not mentioned in prep.mk.
Hope that was okay. 

In general that does not seem consistent to me as some prep-*.mk do not have a call-... function defined...

Thanks for your help.

File: make-rules/prep.mk [hunk 1]
```diff
@@ -25,11 +25,16 @@
 
 include $(WS_MAKE_RULES)/prep-download.mk
 include $(WS_MAKE_RULES)/prep-hg.mk
+include $(WS_MAKE_RULES)/prep-git.mk
 include $(WS_MAKE_RULES)/prep-svn.mk
 include $(WS_MAKE_RULES)/prep-unpack.mk
 include $(WS_MAKE_RULES)/prep-patch.mk
 
+$(eval $(call eval-download-rules))
```
Comment (pyhalov) [inline make-rules/prep.mk]: Why do we need these? It's already called in prep-download.mk.
Perhaps, missing some solaris-userland commit?

Comment (chincheta0815) [reply make-rules/prep.mk]: Okay. Then I think I understand...
BUT then there should be a change in prep-hg-rules.mk calling the respective function there directly as it is done in prep-download.mk, or not?

File: make-rules/prep.mk [hunk 2]
```diff
@@ -25,11 +25,12 @@
 
 include $(WS_MAKE_RULES)/prep-download.mk
 include $(WS_MAKE_RULES)/prep-hg.mk
+include $(WS_MAKE_RULES)/prep-git.mk
 include $(WS_MAKE_RULES)/prep-svn.mk
 include $(WS_MAKE_RULES)/prep-unpack.mk
 include $(WS_MAKE_RULES)/prep-patch.mk
 
-$(eval $(call eval-hg-rules))
+$(eval $(call eval-git-rules))
```
Comment (pyhalov) [inline make-rules/prep.mk]: Do we still need this?

**Comment by pyhalov** [comment]
Try to build some component using this mk file. This is the only way to find out if it's working ;)

**Comment by chincheta0815** [comment]
Okay. This turned out to work with a component I am working on.
The eval-hg-rules can then also be remove after adapting prep-hg-mk I think...
I will do that in another pr.

**Review by pyhalov — APPROVED** [review]

**Comment by chincheta0815** [comment]
Sorry. Stupid me.

**Comment by chincheta0815** [comment]
@pyhalov @Mno-hime If everything is okay with this PR would you mind merging it. After that I want to PR the "cleanup" for prep-hg.mk.

**Pull request merged**